# constants for which grasps should be filtered
string SIDEWAY_GRASP_FILTER="side"
string TOPDOWN_GRASP_FILTER="top"


# constants for which function should be optimized when selecting a grasp
string RANDOM_SELECTION="random"
string CLOSEST_ORIENTATION_SELECTION="closest_orientation"
string CLOSEST_JOINTSPACE_SELECTION="closest_joint_space"
string CLOSEST_TO_CENTROID_SELECTION="closest_to_centroid"

# constants for which method to use to generate grasps
string HEURISTIC="heuristic_grasp"
string AGILE_GRASP="agile_grasp"

# Goal
sensor_msgs/PointCloud2 cloud_plane
float32[4] cloud_plane_coef
sensor_msgs/PointCloud2[] cloud_clusters
int32 target_object_cluster_index

string grasp_generation_method
string grasp_selection_method
string grasp_filter_method


---
# Result
    bool success
    string error_msg
    sensor_msgs/JointState approach_joint_state
    geometry_msgs/PoseStamped approach_pose
    sensor_msgs/JointState grasp_joint_state
    geometry_msgs/PoseStamped grasp_pose

---
# Feedback
bool status
